ledger/src/check_next_block.rs

```diff
@@ -125,6 +106,9 @@ impl<N: Network, C: ConsensusStorage<N>> Ledger<N, C> {
             ratified_finalize_operations,
         )?;
 
+        // Ensure the certificates in the block subdag have met quorum requirements.
+        self.check_block_subdag_quorum(block)?;
```

Comment:
How would we feel about moving this to the syncing logic? Reasoning:
- This check is only needed when syncing, so it is better separation of concerns.
- This check might actually be expensive, because it deserializes a large amount of addresses from disk, which is a key reason why in snarkOS there is now a `committee_cache`.

Reply:
Ideally it is constricted to `snarkVM` so we do not require splitting consensus checks between repos. However I do see the concern of performance, which is why I inquired about potentially just checking the anchor certificate rather than all of the certificates and relying on Bullshark properties to guarantee safety.

Open to other ideas

Reply:
It lives: https://github.com/ProvableHQ/snarkVM/pull/2626